contracts/vaults/BaseVaults/base/RibbonVault.sol

```diff
@@ -591,11 +591,22 @@ contract RibbonVault is
         uint256 currentRound = vaultState.round;
         roundPricePerShare[currentRound] = newPricePerShare;
 
+        uint256 lastQueuedWithdrawAmount =
+            uint256(vaultState.lastQueuedWithdrawAmount);
+
         // Take management / performance fee from previous round and deduct
         lockedBalance = _lockedBalance.sub(
-            _collectVaultFees(_lockedBalance.add(queuedWithdrawAmount))
+            _collectVaultFees(
+                _lockedBalance.add(
+                    queuedWithdrawAmount > lastQueuedWithdrawAmount
+                        ? queuedWithdrawAmount.sub(lastQueuedWithdrawAmount)
+                        : 0
```

Comment:
Probably makes sense to make this a variable like `withdrawAmountDiff` to help with readability